scheduler/pmc-processor-task/src/main/kotlin/ac/uk/ebi/pmc/PmcProcessorApp.kt

```diff
@@ -12,22 +13,29 @@ import java.io.File
 
 @SpringBootApplication
 class PmcProcessorApp {
+
     @Bean
-    fun taskExecutor(properties: PmcImporterProperties, importer: PmcBatchImporter, submitter: PmcBatchSubmitter) =
-        TaskExecutor(properties, importer, submitter)
+    fun taskExecutor(
+        properties: PmcImporterProperties,
+        importer: PmcSubProcessor,
+        submitter: PmcBatchSubmitter,
+        pmcSubmissionLoader: PmcSubmissionLoader
+    ) =
+        TaskExecutor(properties, importer, submitter, pmcSubmissionLoader)
 }
 
 class TaskExecutor(
     private val properties: PmcImporterProperties,
-    private val batchImporter: PmcBatchImporter,
-    private val pmcBatchSubmitter: PmcBatchSubmitter
+    private val pmcSubProcessor: PmcSubProcessor,
```

Comment:
This is just something I was thinking but, I see that we use indistinctively "Sub" and "Submission" to refer to the same thing. We should pick one or the other and stick to it (I'd prefer if we go with the latter).

I just say this because, for example, in this case, this variable called `pmcSubProcessor` might refer to a processor for PMC submissions but it might also be interpreted as a PMC sub-process (as in a process that's part of another process).

The same thing could apply for other parts of the code. I know I might be overcomplicating here but better take this decision now that we're refactoring 😅